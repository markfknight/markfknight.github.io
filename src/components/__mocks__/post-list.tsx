import React, { ReactElement } from 'react';

const PostList = (): ReactElement => <></>;

export { PostList };
