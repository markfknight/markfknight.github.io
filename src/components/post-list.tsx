import { graphql, Link, useStaticQuery } from 'gatsby';
import React from 'react';

export type PostListData = {
  allMarkdownRemark: {
    totalCount: number;
    edges: [
      {
        node: {
          id: string;
          fields: {
            slug: string;
          };
          frontmatter: {
            title: string;
            date: string;
          };
          excerpt: string;
        };
      }
    ];
  };
};

export type PostListProps = {
  title?: string;
};

export const PurePostList: React.FC<PostListProps & { data: PostListData }> = (
  props
) => (
  <>
    <div className="space-y-8">
      {props.data.allMarkdownRemark.edges.map(({ node }) => (
        <article
          className="p-4 flow-root rounded-sm shadow-md space-x-2"
          key={node.id}
        >
          <header>
            <h2>
              <Link to={node.fields.slug}>{node.frontmatter.title}</Link>
            </h2>
            <span className="text-gray-500">{node.frontmatter.date}</span>
          </header>
          <section>
            <p>{node.excerpt}</p>
          </section>
        </article>
      ))}
    </div>
  </>
);

export const PostList: React.FC<PostListProps> = (props) => {
  const data: PostListData = useStaticQuery(
    graphql`
      query {
        allMarkdownRemark(sort: { fields: [frontmatter___date], order: DESC }) {
          totalCount
          edges {
            node {
              id
              excerpt
              fields {
                slug
              }
              frontmatter {
                title
                date(formatString: "DD MMMM, YYYY")
              }
            }
          }
        }
      }
    `
  );
  return <PurePostList {...props} data={data}></PurePostList>;
};
