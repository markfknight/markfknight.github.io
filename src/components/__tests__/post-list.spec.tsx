import React, { PropsWithChildren } from 'react';
import { cleanup, render } from '@testing-library/react';

import { PostListProps, PostListData, PurePostList } from '../post-list';

afterEach(cleanup);

describe(`Post List snapshot`, () => {
  it(`renders correctly`, () => {
    const props: PropsWithChildren<PostListProps & { data: PostListData }> = {
      data: {
        allMarkdownRemark: {
          totalCount: 1,
          edges: [
            {
              node: {
                id: `id`,
                fields: {
                  slug: `slug`,
                },
                frontmatter: {
                  title: `title`,
                  date: `2021-01-01`,
                },
                excerpt: `excerpt`,
              },
            },
          ],
        },
      },
    };

    const tree = render(<PurePostList {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
