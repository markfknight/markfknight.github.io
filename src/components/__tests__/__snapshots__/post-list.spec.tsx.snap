// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post List snapshot renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="space-y-8"
      >
        <article
          class="p-4 flow-root rounded-sm shadow-md space-x-2"
        >
          <header>
            <h2>
              <a
                href="slug"
              >
                title
              </a>
            </h2>
            <span
              class="text-gray-500"
            >
              2021-01-01
            </span>
          </header>
          <section>
            <p>
              excerpt
            </p>
          </section>
        </article>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="space-y-8"
    >
      <article
        class="p-4 flow-root rounded-sm shadow-md space-x-2"
      >
        <header>
          <h2>
            <a
              href="slug"
            >
              title
            </a>
          </h2>
          <span
            class="text-gray-500"
          >
            2021-01-01
          </span>
        </header>
        <section>
          <p>
            excerpt
          </p>
        </section>
      </article>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
